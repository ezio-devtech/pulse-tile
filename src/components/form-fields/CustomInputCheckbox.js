import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class CustomInputCheckbox extends PureComponent {
    static propTypes = {
      label: PropTypes.string,
      input: PropTypes.object.isRequired,
      meta: PropTypes.object,
    };

    render() {
      const { label, input, id } = this.props;

      return (
        <div className="wrap-fcustominp">
          <div className="fcustominp">
            <input
              type="checkbox"
              checked={input.value}
              id={id || ''}
              {...input}
            />
            <label htmlFor={input.name} />
          </div>
          {label ? <label htmlFor={input.name} className="fcustominp-label">{label}</label> : null}
        </div>
      )
    }
}
