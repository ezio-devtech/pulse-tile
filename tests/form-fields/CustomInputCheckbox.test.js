import React from 'react';
// import renderer from 'react-test-renderer';
import {configure, shallow, mount} from 'enzyme'
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-15';
configure({ adapter: new Adapter() });

import CustomInputCheckbox from '../../src/components/form-fields/CustomInputCheckbox';

const id = 'test-id';
const label = 'Test label';
const input = {name: 'test-name', value: true};

describe('Component <CustomInputCheckbox />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
        <CustomInputCheckbox
          testProps='testProps'
          id={id}
          label={label}
          input={input}
        />);
    expect(component.find('input')).toHaveLength(1);
    expect(component.find('label')).toHaveLength(2);
    expect(component.instance().props['label']).toEqual(label);
    expect(component.instance().props['id']).toEqual(id);
    expect(component.instance().props['input']).toEqual(input);

    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  it('should renders without label prop correctly', () => {
    const component = shallow(
        <CustomInputCheckbox
          id={id}
          input={input}
        />);

    expect(component.find('input')).toHaveLength(1);
    expect(component.find('label')).toHaveLength(1);
    expect(component.instance().props['label']).toEqual(undefined);

    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  it('should renders without id prop correctly', () => {
    const component = shallow(
        <CustomInputCheckbox
          label={label}
          input={input}
        />);

    expect(component.find('input')).toHaveLength(1);
    expect(component.instance().props['id']).toEqual(undefined);

    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  // it('changes the text after click', () => {
  //   // Render a checkbox with label in the document
  //   const component = mount(
  //     <CustomInputCheckbox
  //       id={id}
  //       input={input}
  //     />);
  //
  //   const inputEl = component.find('input');
  //   const inputEl = () => component.find('input');
  //
  //   expect(inputEl().props().checked).toEqual(true);
  //   inputEl().toNotBe.checked();
  //   inputEl().simulate('change', {target: {checked: true}});
  //   expect(inputEl().props().checked).toEqual(false);
  //   var checkbox = () => wrapper.find('input');
  //   checkbox().should.not.be.checked();
  //   checkbox().simulate('change', {target: {checked: true}});
  // });
});
