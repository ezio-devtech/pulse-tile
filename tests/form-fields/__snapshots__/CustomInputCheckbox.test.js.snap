// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <CustomInputCheckbox /> should renders with all props correctly 1`] = `
<div
  className="wrap-fcustominp"
>
  <div
    className="fcustominp"
  >
    <input
      checked={true}
      id="test-id"
      name="test-name"
      type="checkbox"
      value={true}
    />
    <label
      htmlFor="test-id"
    />
  </div>
  <label
    className="fcustominp-label"
    htmlFor="test-id"
  >
    Test label
  </label>
</div>
`;

exports[`Component <CustomInputCheckbox /> should renders without id prop correctly 1`] = `
<div
  className="wrap-fcustominp"
>
  <div
    className="fcustominp"
  >
    <input
      checked={true}
      id=""
      name="test-name"
      type="checkbox"
      value={true}
    />
    <label
      htmlFor=""
    />
  </div>
  <label
    className="fcustominp-label"
    htmlFor=""
  >
    Test label
  </label>
</div>
`;

exports[`Component <CustomInputCheckbox /> should renders without label prop correctly 1`] = `
<div
  className="wrap-fcustominp"
>
  <div
    className="fcustominp"
  >
    <input
      checked={true}
      id="test-id"
      name="test-name"
      type="checkbox"
      value={true}
    />
    <label
      htmlFor="test-id"
    />
  </div>
</div>
`;
